File: left_rotate_by_d.py
def reverse(arr,low,high):
    while low < high:
        arr[low],arr[high]=arr[high],arr[low]
        low+=1
        high-=1
    return arr

def left_rotate(arr,n,d):
    reverse(arr,0,d-1)
    reverse(arr,d,n-1)
    reverse(arr,0,n-1)
    return arr

print(left_rotate([1,2,3,4,5,6,7,8,9],9,2))

File: left_rotate_by_d.cpp
#include <iostream>
using namespace std;

void reverse(int arr[],int low, int high)
{
    while (low < high){
        int temp = arr[low];
        arr[low] = arr[high];
        arr[high] = temp;
        low ++;
        high --;
    }
}

void rotate(int arr[],int n, int d) 
{
    reverse(arr,0,d-1);
    reverse(arr,d,n-1);
    reverse(arr,0,n-1);
}

int main()
{
    int arr[] = {1,2,3,4,5};
    rotate(arr,5,2);
    for (int i=0; i< 5; i++)
        cout << arr[i] << " ";
    return 0;
}